app/feature/feature-delete-account/src/main/kotlin/com/hedvig/android/feature/deleteaccount/DeleteAccountDestination.kt

```diff
@@ -0,0 +1,199 @@
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.ButtonColors
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.datasource.CollectionPreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.halilibo.richtext.markdown.Markdown
+import com.halilibo.richtext.ui.material3.RichText
+import com.hedvig.android.core.designsystem.component.button.HedvigContainedButton
+import com.hedvig.android.core.designsystem.component.error.HedvigErrorSection
+import com.hedvig.android.core.designsystem.component.progress.HedvigFullScreenCenterAlignedProgress
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.ui.scaffold.HedvigScaffold
+import com.hedvig.android.feature.chat.DeleteAccountViewModel
+import hedvig.resources.R
+
+@Composable
+internal fun DeleteAccountDestination(
+  viewModel: DeleteAccountViewModel,
+  navigateUp: () -> Unit,
+  navigateBack: () -> Unit,
+) {
+  val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+  DeleteAccountScreen(
+    uiState = uiState,
+    navigateUp = navigateUp,
+    navigateBack = navigateBack,
+    retryLoading = { viewModel.emit(DeleteAccountEvent.RetryLoading) },
+    initiateAccountDeletion = { viewModel.emit(DeleteAccountEvent.InitiateAccountDeletion) },
+  )
+}
+
+@Composable
+private fun DeleteAccountScreen(
+  uiState: DeleteAccountUiState,
+  navigateUp: () -> Unit,
+  navigateBack: () -> Unit,
+  retryLoading: () -> Unit,
+  initiateAccountDeletion: () -> Unit,
+) {
+  HedvigScaffold(navigateUp = navigateUp) {
+    when (uiState) {
+      DeleteAccountUiState.FailedToLoadDeleteAccountState -> {
+        HedvigErrorSection(retry = retryLoading, Modifier.weight(1f))
+      }
+
+      DeleteAccountUiState.Loading -> {
+        HedvigFullScreenCenterAlignedProgress(Modifier.weight(1f))
+      }
+
+      is DeleteAccountUiState.Success -> {
+        when (uiState) {
+          is DeleteAccountUiState.Success.CanDelete -> {
+            if (uiState.failedToPerformDeletion) {
+              HedvigErrorSection(retry = retryLoading, Modifier.weight(1f))
+            } else {
+              DeleteScreenContents(
+                title = stringResource(R.string.DELETE_ACCOUNT_DELETE_ACCOUNT_TITLE),
```

Comment:
Could you add extension functions on `DeleteAccountUiState.Success` and do eg. `uiState.title`, `uiState.description` etc. to avoid having these when clauses?

Reply:
100% agree. Addressed here
https://github.com/HedvigInsurance/android/commit/9d535744ffc17c0eafbfe2cbab3095df1b7ba13c